### PR TITLE
test(drive-abci): improve batch validation test coverage

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## Issue being fixed or feature implemented

Improve test coverage for `packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/batch/` module, targeting the lowest-coverage submodules identified by Codecov:
- action_validation (72%, 744 missed lines)
- state/v0 (74%, 127 missed lines)
- data_triggers (78%, 260 missed lines)

## What was done?

Added 14 new integration tests across 3 new test modules that exercise previously uncovered validation error paths:

**`emergency_action` module (5 tests):**
- Successful pause operation
- Pause when already paused returns `TokenAlreadyPausedError`
- Resume when not paused returns `TokenNotPausedError`
- Full pause-and-resume lifecycle
- Unauthorized emergency action returns `UnauthorizedTokenActionError`

**`destroy_frozen_funds` module (3 tests):**
- Destroying unfrozen account funds returns `IdentityTokenAccountNotFrozenError`
- Unauthorized destruction returns `UnauthorizedTokenActionError`
- Successful freeze-then-destroy lifecycle by contract owner

**`additional_validation` module (6 tests):**
- Freezing already-frozen account returns `IdentityTokenAccountAlreadyFrozenError`
- Unfreezing non-frozen account returns `IdentityTokenAccountNotFrozenError`
- Minting to frozen account (when disallowed) returns `IdentityTokenAccountFrozenError`
- Burn with insufficient balance returns `IdentityDoesNotHaveEnoughTokenBalanceError`
- Transfer with insufficient balance returns `IdentityDoesNotHaveEnoughTokenBalanceError`
- Transfer from frozen sender returns `IdentityTokenAccountFrozenError`

These tests cover validation paths in `token_emergency_action_transition_action/state_v0`, `token_destroy_frozen_funds_transition_action/state_v0`, `token_freeze_transition_action/state_v0`, `token_unfreeze_transition_action/state_v0`, `token_mint_transition_action/state_v0`, `token_burn_transition_action/state_v0`, and `token_transfer_transition_action/state_v0`.

## How Has This Been Tested?

All 14 new tests pass locally:
```
cargo test -p drive-abci --lib "batch::tests::token"
test result: ok. 154 passed; 0 failed; 5 ignored; 0 measured; 1647 filtered out
```

No existing tests were broken by these changes.

## Breaking Changes

None. This PR only adds new test code.

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Expanded token test coverage for validation flows: freeze/unfreeze, minting, burning, transfers, and edge cases (insufficient balance, frozen-account rules).
  * Added tests covering destruction of frozen funds, verifying authorization and correct error on unfrozen accounts.
  * Added emergency action tests for pause/resume behavior, including repeated action errors and owner authorization checks.

### CODE REVIEW

File: packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/batch/tests/token/additional_validation/mod.rs [hunk 1]
```diff
@@ -0,0 +1,674 @@
+use super::*;
+
+mod additional_validation_tests {
+    use super::*;
+    use dpp::tokens::info::v0::IdentityTokenInfoV0Accessors;
+
+    #[test]
+    fn test_token_freeze_already_frozen_should_fail() {
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/batch/tests/token/additional_validation/mod.rs:8]: The existing tests in this directory all use `test_token_*` naming (e.g. `test_token_burn`, `test_token_freeze`, `test_token_mint_by_owner_allowed_sending_to_self`). Our tests follow the same convention. No rename needed.
